### PR TITLE
Migrate notice creation view/client logic to engine.

### PR DESCRIPTION
**Pull request by eschlange**
* Moved in toast notice dependencies into engine.
* Updated controller signatures to enable/disable notices.
* Updated controller logic to use new flag.
* Updated manifests for new dependencies.

[#95080280]

### CODE REVIEW

File: app/views/social_networking/goals/tool.html.erb [hunk 1]
```diff
@@ -107,4 +107,6 @@
 <script>
   angular.module('socialNetworking.values').value('participantStudyEndDate', '<%= current_participant.active_membership_end_date %>');
   angular.module('socialNetworking.values').value('currentGoals', <%= goals.to_json.html_safe %>);
+  angular.module('socialNetworking.values').value('noticesEnabled', <%= Rails.configuration.enable_notices %>);
+  angular.module('socialNetworking.values').value('noticeUtility', Notice);
```
Comment (ericcf) [inline app/views/social_networking/goals/tool.html.erb]: Are you using `Notice` in other places? Is that why you're injecting it as a global variable?

File: app/assets/javascripts/notices.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
+var Notice = (function() {
```
Comment (ericcf) [inline app/assets/javascripts/notices.js]: Where are you testing this logic?

Comment (eschlange) [inline app/assets/javascripts/notices.js]: Yes, I have yet to check it in. I'm still working on a few tests for the udpates to the angular controllers.

**Comment by eschlange** [comment]
@ericcf Hi Eric, I updated the JS code to follow Lint standards and added a few Javascript test cases for the Notices utility. Once this is merged, I'll integrate this into the SunnySide host application.

**Comment by ericcf** [comment]
Thanks!

**Pull request merged**